internal/controller/entra/main.go

```diff
@@ -57,21 +57,48 @@ func New() *Controller {
 	return controller
 }
 
+func EntraUsernameForExternalEmail(email string) (string, error) {
+	if config.EntraTenantPrimaryDomain() == "" {
```

Comment:
As far as my understanding of Go goes, the uppercase exports the function. But it looks like this function is only used by within this file and for a test. Is exporting a function for a test typical in Go?